nitrocli/src/options.rs

```diff
@@ -0,0 +1,155 @@
+// options.rs
+
+// *************************************************************************
+// * Copyright (C) 2017-2018 Daniel Mueller (deso@posteo.net)              *
+// *                                                                       *
+// * This program is free software: you can redistribute it and/or modify  *
+// * it under the terms of the GNU General Public License as published by  *
+// * the Free Software Foundation, either version 3 of the License, or     *
+// * (at your option) any later version.                                   *
+// *                                                                       *
+// * This program is distributed in the hope that it will be useful,       *
+// * but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+// * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+// * GNU General Public License for more details.                          *
+// *                                                                       *
+// * You should have received a copy of the GNU General Public License     *
+// * along with this program.  If not, see <http://www.gnu.org/licenses/>. *
+// *************************************************************************
+
+use std::fmt;
+use std::io;
+use std::str::FromStr;
+
+use crate::commands;
+use crate::error::Error;
+use crate::Result;
+
+/// A top-level command for nitrocli.
+#[derive(Debug)]
+pub enum Command {
+  Clear,
+  Close,
+  Open,
+  Status,
+}
+
+impl Command {
+  /// Execute this command with the given arguments.
+  pub fn execute(&self, args: Vec<String>) -> Result<()> {
+    match *self {
+      Command::Clear => clear(args),
+      Command::Close => close(args),
+      Command::Open => open(args),
+      Command::Status => status(args),
+    }
+  }
+}
+
+impl fmt::Display for Command {
+  fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+    write!(
+      f,
+      "{}",
+      match *self {
+        Command::Clear => "clear",
+        Command::Close => "close",
+        Command::Open => "open",
+        Command::Status => "status",
+      }
+    )
+  }
+}
+
+impl FromStr for Command {
+  type Err = ();
+
+  fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
```

Comment:
It appears we are using:
1) `use std::str::FromStr` (full qualifier in `use` statement)
2) `use std::io` and
3) `std::result::Result` (full qualifier at site of use)

I'd rather we agree on one way to reference symbols. In the past I've typically used the first variant, but it seems uncommon in the community I admit that it may not be the best one. I think 2) is most common. Shall we converge on that?